frontend/src/metabase/visualizations/visualizations/Table.jsx

```diff
@@ -361,6 +361,9 @@ export default class Table extends Component {
         data: {
           cols: columnIndexes.map(i => cols[i]),
           rows: rows.map(row => columnIndexes.map(i => row[i])),
+          resultsMetadataColumns: columnIndexes.map(
+            i => results_metadata.columns[i],
+          ),
```

Comment:
It's a bummer, I need to set this in the state, but I wouldn't be able to get the desired column metadata elsewhere since this code block took care of the visible columns (via `columnIndexes`)